src/PayzenGatewayFactory.php

```diff
@@ -42,7 +55,7 @@ protected function populateConfig(ArrayObject $config)
             'payum.action.refund'          => new Action\RefundAction(),
             'payum.action.status'          => new Action\StatusAction(),
             'payum.action.notify'          => new Action\NotifyAction(),
-            'payum.action.api.request'     => new Action\Api\ApiRequestAction(),
+            'payum.action.api.request'     => new Action\Api\ApiRequestAction($this->transactionId),
```

Comment:
Inject transaction id generator into Api instance (see next comment) instead of ApiRequestAction .
ApiRequestAction is aware of Api (just give access to generator).